neqo-transport/src/packet/mod.rs

```diff
@@ -611,69 +614,73 @@ impl<'a> PublicPacket<'a> {
     ///
     /// This will return an error if the packet could not be decoded.
     #[allow(clippy::similar_names)]
-    pub fn decode(data: &'a [u8], dcid_decoder: &dyn ConnectionIdDecoder) -> Res<(Self, &'a [u8])> {
+    pub fn decode(
+        data: &'a mut [u8],
+        dcid_decoder: &dyn ConnectionIdDecoder,
+    ) -> Res<(Self, &'a mut [u8])> {
         let mut decoder = Decoder::new(data);
-        let first = Self::opt(decoder.decode_uint::<u8>())?;
+        let first = PublicPacket::opt(decoder.decode_uint::<u8>())?;
```

Comment:
Am I to guess that this needs to be `PublicPacket` and not `Self` because the lifetimes are different?  Subtle shit like that is where rust loses points in my book.

Reply:
I was able to revert all but one of those changes. Reverting the one that is left makes the borrow checker freak out.